llvm-15.0.3/llvm-crash-analyzer/lib/Decompiler/Decompiler.cpp

```diff
@@ -117,12 +133,6 @@ MachineInstr *crash_analyzer::Decompiler::addInstr(
 
   auto TII = MF->getSubtarget().getInstrInfo();
   auto TRI = MF->getSubtarget().getRegisterInfo();
-  // No need for the NOOPs within MIR representation.
-  // TODO: Optimize this. Can we check if it is a noop from MCID?
```

Comment:
I am wondering, what would happen if we encounter a NOOP during Taint Analysis, which is not the crash-start for that function.
Have you checked it, should we skip such instructions?

Reply:
That is a valid point but is outside of the context of this change. I will look into that separately. Thanks